### DISCUSSION TITLE
ManyHasMany table example

### DISCUSSION THREAD

**Discussion by ghost** [nextras/orm#538] [Q&A]
Hi, I am wondering how ManyHasMany table looks like.

We know that we have to create primary_x_secondary table but we don't know which columns it should have. Does anybody have an example of SQL or something like that?

We are trying to solve it for about 2 hours.

Users class
` * @property Eshop[]|null $eshops {m:m Eshop::$users}`

Eshop class
` * @property User[]|null $users {m:m User::$eshops, isMain=true}`

**Comment by hrach** [comment]
It should have:
- two columns, both creating a composite primary key
- each of them being a foreign key
- the column names do not matter much

Resolution logic may be found here: https://github.com/nextras/orm/blob/336140091ecc6053ec5161b3f48e2803f13be3ad/src/Mapper/Dbal/Conventions/Conventions.php#L315

**Reply by ghost** [reply]
I tried exactly the same steps but when we try to get data it response with empty array.

**Reply by hrach** [reply]
did you clear cache?

**Reply by ghost** [reply]
Yes I did. But still nothing coming up.

**Reply by ghost** [reply]
And when we try to add something into $eshops it throws an error.

`No primary keys detected for many has many 'product_x_eshop' join table.`

But we don't have ManyHasMany anywhere else.

**Reply by hrach** [reply]
Show me your table definition please.

**Reply by ghost** [reply]
```sql
create table touchlogis.user_x_eshop
(
    user  int not null,
    eshop int not null,
    primary key (user, eshop),
    constraint user_x_eshop_eshop
        foreign key (eshop) references touchlogis.eshop (id),
    constraint user_x_eshop_user
        foreign key (user) references touchlogis.user (id)
);

create index user_x_eshop_eshop_idx
    on touchlogis.user_x_eshop (eshop);
```

**Reply by hrach** [reply]
Well, the error says you look for 'product_x_eshop' join table. Is it just the same issue with another relationship or is there some other issue?

**Reply by ghost** [reply]
Yes but there is a problem there is no ManyHasMany relationship between Eshop and Product see images
This one is from Eshop
<img width="530" alt="Snímek obrazovky 2021-10-08 v 12 52 54" src="https://user-images.githubusercontent.com/75794111/136544568-1f8ea513-ccaf-4c30-b8a9-f2e9741fe6a5.png">

There is Product
<img width="529" alt="Snímek obrazovky 2021-10-08 v 12 53 52" src="https://user-images.githubusercontent.com/75794111/136544673-8f6cce80-80d3-4f39-a913-39f6df9e210e.png">
